### PR TITLE
provide way of correlating eyes tests with particular runs

### PR DESCRIPTION
**Pull request by Bjvanminnen**
image

Use BatchInfo to provide additional metadata about a test, allowing us to correlate which eyes tests are from the same run and which arent.

### CODE REVIEW

**Comment by Bjvanminnen** [comment]
(rebased on eslintUpgrade to hopefully get a clean run)

File: dashboard/test/ui/step_definitions/eyes_steps.rb [hunk 1]
```diff
@@ -6,6 +6,10 @@
 
 When(/^I open my eyes to test "([^"]*)"$/) do |test_name|
   ensure_eyes_available
+
+  batch_name = test_name + " | " + ENV['BATCH_NAME']
+  @eyes.batch = Applitools::Base::BatchInfo.new(batch_name)
```
Comment (islemaster) [inline dashboard/test/ui/step_definitions/eyes_steps.rb:11]: I think I misunderstood the 'batch' capabilities at first. Will it actually group a whole run for us - where we can see individual tests by expanding on the page? ([source](http://support.applitools.com/customer/portal/articles/1665101-aggregating-grouping-several-tests-under-the-same-batch-))

image

If so, that seems better.  To get that working, you might need to pass the second `started_at` parameter to `BatchInfo.new` ([see here](https://docs.omniref.com/ruby/gems/eyes_selenium/1.23.0/symbols/Applitools::BatchInfo)) as I suspect the start time is really the key for these.

Comment (islemaster) [inline dashboard/test/ui/step_definitions/eyes_steps.rb:11]: Unless we really want to keep the tests as separate batches, and use the batch functionality to test different platforms/resolutions, as they do in the screenshot.

**Comment by islemaster** [comment]
Have you confirmed that baselines are used correctly across runs? It's not obvious from your screenshot.

File: dashboard/test/ui/step_definitions/eyes_steps.rb (hunk 1 above)
Comment (Bjvanminnen) [inline dashboard/test/ui/step_definitions/eyes_steps.rb:11]: My first attempt had us using a single batch per instance of eyes. It still resulted in a single top level item per test (i.e. it did not show multiple items like in the screenshot you have). I was not using the started_at param of BatchInfo.new, which maybe makes a difference. I can explore this a little further.

**Comment by Bjvanminnen** [comment]
Not really. I'm not sure there's any great way to do this other than to push to test.

**Comment by Bjvanminnen** [comment]
Tried passing in the start time, and still ended up with separate items for separate tests. Not sure why this is different from the screenshot you shared.

image

**Comment by islemaster** [comment]
It's probably because we're creating a new `BatchInfo` object for each test (the documentation does call out that you should only set this once per batch).  I'm not sure there's any easy way around that, given our current runner setup.  I'd love to explore that someday, but not necessary now unless you're interested.  Current solution LGTM.

**Comment by Bjvanminnen** [comment]
With my changes I was creating the BatchInfo in `ensure_eyes_available`, such that it should have been once per instantiation of `./runner`.

**Pull request merged**